Lib/inspect.py

```diff
@@ -1638,7 +1638,30 @@ def getclosurevars(func):
 
 # -------------------------------------------------- stack frame extraction
 
-Traceback = namedtuple('Traceback', 'filename lineno function code_context index')
+_Traceback = namedtuple('_Traceback', 'filename lineno function code_context index')
+
+class Traceback(_Traceback):
+    def __new__(cls, filename, lineno, function, code_context, index, *, positions=None):
+        instance = super().__new__(cls, filename, lineno, function, code_context, index)
+        instance.positions = positions
+        return instance
+    
+    def __repr__(self):
```

Comment:
We should also override `__eq__` (and maybe `__hash__`) to take `.positions` into account.

Reply:
I didn't change this because that is backwards-incompatible if people are already comparing the tuples :(

Reply:
Right, good point.